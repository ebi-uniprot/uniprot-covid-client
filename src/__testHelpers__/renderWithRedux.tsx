/* eslint-disable */
import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { createMemoryHistory, MemoryHistory } from 'history';
import rootReducer from '../state/rootReducer';
import { Router, Route } from 'react-router-dom';

type RenderOptions = {
  route?: string;
  history?: MemoryHistory<any>;
  path?: string;
  initialState?: any;
  store?: any;
};

const renderWithRedux = (
  ui: React.Component,
  {
    route = '',
    history = createMemoryHistory({ initialEntries: [route] }),
    path,
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
  }: RenderOptions = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>
          {path ? <Route path={path} render={() => ui} /> : ui}
        </Router>
      </Provider>
    ),
    store,
    history,
  };
};

export default renderWithRedux;
