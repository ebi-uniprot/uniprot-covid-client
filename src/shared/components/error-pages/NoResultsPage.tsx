import React from 'react';
import { Message } from 'franklin-sites';
import ErrorPage from './ErrorPage';
import ArtWork from './svgs/no-results-found.svg';

const ErrorMessage = () => (
  <Message level="info">
    <h4>Sorry, no results were found!</h4>
    <span>Please try a different criteria</span>
  </Message>
);

const NoResultsPage = () => (
  <ErrorPage
    artwork={<ArtWork />}
    message={<ErrorMessage />}
    testid="no-results-page"
  />
);

export default NoResultsPage;
