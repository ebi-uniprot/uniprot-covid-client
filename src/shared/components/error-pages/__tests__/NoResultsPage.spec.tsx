import React from 'react';
import NoResultsPage from '../NoResultsPage';
import renderWithRedux from '../../../../shared/__test-helpers__/RenderWithRedux';

describe('NoResultsPage component', () => {
  test('should render', () => {
    const { asFragment } = renderWithRedux(<NoResultsPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
