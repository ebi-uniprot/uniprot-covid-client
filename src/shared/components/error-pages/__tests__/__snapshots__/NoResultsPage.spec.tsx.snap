// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResultsPage component should render 1`] = `
<DocumentFragment>
  <div
    class="error-page-container"
    data-testid="no-results-page"
  >
    <test-file-stub
      classname="error-page-container__art-work"
    />
    <div
      class="message message--info"
      role="status"
    >
      <div
        class="message__side-border"
      />
      <svg
        fill="currentColor"
        height="1.125em"
        viewBox="0 0 18 18"
        width="1.125em"
      >
        <path
          clip-rule="evenodd"
          d="M18 9A9 9 0 110 9a9 9 0 0118 0zM8 4a1 1 0 112 0v6a1 1 0 11-2 0V4zm1 9a1 1 0 100 2 1 1 0 000-2z"
          fill-rule="evenodd"
        />
      </svg>
      <section
        class="message__content"
      >
        <small>
          <h4>
            Sorry, no results were found!
          </h4>
          <span>
            Please try a different criteria
          </span>
        </small>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
